doc/source/serve/doc_code/batching_guide.py

```diff
@@ -33,3 +34,65 @@ async def __call__(self, multiple_samples: List[int]) -> List[int]:
 handle = serve.run(Model.bind())
 assert ray.get([handle.remote(i) for i in range(8)]) == [i * 2 for i in range(8)]
 # __batch_end__
+
+
+# __single_stream_begin__
+import asyncio
+from typing import AsyncGenerator
+from starlette.requests import Request
+from starlette.responses import StreamingResponse
+
+from ray import serve
+
+
+@serve.deployment
+class StreamingResponder:
+    async def generate_numbers(self, max: int) -> AsyncGenerator[int, None]:
+        for i in range(max):
+            yield i
+            asyncio.sleep(0.1)
+
+    def __call__(self, request: Request) -> StreamingResponse:
+        max = int(request.query_params.get("max", "25"))
+        gen = self.generate_numbers(max)
+        return StreamingResponse(gen, status_code=200, media_type="text/plain")
+
+
+# __single_stream_end__
+
+# TODO (shrekris-anyscale): add unit tests
+
+# __batch_stream_begin__
+import asyncio
+from typing import List, AsyncGenerator
+from starlette.requests import Request
+from starlette.responses import StreamingResponse
+
+from ray import serve
+
+
+@serve.deployment
+class StreamingResponder:
+    @serve.batch(max_batch_size=5, batch_wait_timeout_s=15)
+    async def generate_numbers(
+        self, max_list: List[int]
+    ) -> AsyncGenerator[List[int], None]:
+        for i in range(max(max_list)):
+            next_numbers = []
+            for requested_max in max_list:
+                if requested_max > i:
+                    next_numbers.append(i)
+                else:
+                    next_numbers.append(StopIteration)
+            yield next_numbers
```

Comment:
```suggestion
            yield [
                i for i, requested_max in enumerate(max_list) if i <= requested_max else StopIteration
            ]
```
might be easier to read? not sure

Reply:
This feels harder to read in my opinion. I'd prefer to stick with the explicit `if`-`else`.